src/Resources/contao/dca/tl_user.php

```diff
@@ -985,4 +1019,155 @@ public function toggleVisibility($intId, $blnVisible, DataContainer $dc=null)
 
 		$objVersions->create();
 	}
+
+	/**
+	 * Builds the palette for 2FA dynamically.
+	 *
+	 * @param DataContainer $dc
+	 */
+	public function build2faPalette(DataContainer $dc)
+	{
+		$activeRecord = UserModel::findById((int) $dc->id);
+
+		if ($activeRecord === null)
+		{
+			return;
+		}
+
+		foreach ($GLOBALS['TL_DCA']['tl_user']['palettes'] as $palette => $v)
+		{
+			if ($palette === '__selector__')
+			{
+				continue;
+			}
+
+			// Don't show 2FA options, if it is not the logged in user itself and 2FA is disabled.
+			if ($dc->id !== $this->User->id && (int) $activeRecord->use2fa !== '1')
+			{
+				continue;
+			}
+
+			$GLOBALS['TL_DCA']['tl_user']['palettes'][$palette] = str_replace
```

Comment:
Why not use the `PaletteManipulator` here? Would make things easier 😄

Reply:
Definitely! 😎